src/compiler/parser.ts

```diff
@@ -2365,33 +2365,37 @@ namespace ts {
             return false;
         }
 
-        function parsingContextErrors(context: ParsingContext): DiagnosticMessage {
+        function parsingContextErrors(context: ParsingContext): [DiagnosticMessage, string?] {

```

Comment:
Nit: I kinda dislike allocating an array on each of these, but maybe it's fine.

Reply:
Can do!